homeassistant/components/renault/sensor.py

```diff
@@ -0,0 +1,277 @@
+
+
+class RenaultBatteryLevelSensor(RenaultBatteryDataEntity, SensorEntity):
+    """Battery Level sensor."""
+
+    _attr_device_class = DEVICE_CLASS_BATTERY
+    _attr_unit_of_measurement = PERCENTAGE
+
+    @property
+    def state(self) -> int | None:
+        """Return the state of this entity."""
+        return self.data.batteryLevel if self.data else None
+
+    @property
+    def icon(self) -> str:
+        """Icon handling."""
+        return icon_for_battery_level(
+            battery_level=self.state, charging=self.is_charging
+        )
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return the state attributes of this entity."""
+        attrs = super().extra_state_attributes
+        attrs[ATTR_BATTERY_AVAILABLE_ENERGY] = (
+            self.data.batteryAvailableEnergy if self.data else None
+        )
+        return attrs
+
+
+class RenaultBatteryTemperatureSensor(RenaultBatteryDataEntity, SensorEntity):
+    """Battery Temperature sensor."""
+
+    _attr_device_class = DEVICE_CLASS_TEMPERATURE
+    _attr_unit_of_measurement = TEMP_CELSIUS
+
+    @property
+    def state(self) -> int | None:
+        """Return the state of this entity."""
+        return self.data.batteryTemperature if self.data else None
+
+
+class RenaultChargeModeSensor(RenaultChargeModeDataEntity, SensorEntity):
+    """Charge Mode sensor."""
+
+    _attr_device_class = DEVICE_CLASS_CHARGE_MODE
+
+    @property
+    def state(self) -> str | None:
+        """Return the state of this entity."""
+        return self.data.chargeMode if self.data else None
+
+    @property
+    def icon(self) -> str:
+        """Icon handling."""
+        if self.data and self.data.chargeMode == "schedule_mode":
+            return "mdi:calendar-clock"
+        return "mdi:calendar-remove"
+
+
+class RenaultChargeStateSensor(RenaultBatteryDataEntity, SensorEntity):
+    """Charge State sensor."""
+
+    _attr_device_class = DEVICE_CLASS_CHARGE_STATE
+
+    @property
+    def state(self) -> str | None:
+        """Return the state of this entity."""
+        charging_status = self.data.get_charging_status() if self.data else None
+        return slugify(charging_status.name) if charging_status is not None else None
+
+    @property
+    def icon(self) -> str:
+        """Icon handling."""
+        return "mdi:flash" if self.is_charging else "mdi:flash-off"
+
+
+class RenaultChargingRemainingTimeSensor(RenaultBatteryDataEntity, SensorEntity):
+    """Charging Remaining Time sensor."""
+
+    _attr_icon = "mdi:timer"
+    _attr_unit_of_measurement = TIME_MINUTES
+
+    @property
+    def state(self) -> int | None:
+        """Return the state of this entity."""
+        return self.data.chargingRemainingTime if self.data else None
+
+
+class RenaultChargingPowerSensor(RenaultBatteryDataEntity, SensorEntity):
+    """Charging Power sensor."""
+
+    _attr_device_class = DEVICE_CLASS_ENERGY
+    _attr_unit_of_measurement = POWER_KILO_WATT
```

Comment:
Watt is not a compatible unit with device class energy.

https://developers.home-assistant.io/docs/core/entity/sensor#available-device-classes